autowarefoundation/autoware.universe
# feat: system diagnostic monitor

## Description

The diagnostic monitor extends the diagnostic and outputs the top-level operation status of Autoware for MRM. See https://github.com/autowarefoundation/autoware.universe/issues/3829#issuecomment-1702392865 for details.


## Related links

https://github.com/tier4/tier4_autoware_msgs/pull/96
https://github.com/autowarefoundation/autoware.universe/issues/3829#issuecomment-1702392865

## Tests performed

1. Launch planning simulation
2. Launch diagnostics nodes
    - `ros2 launch system_diagnostic_monitor system_diagnostic_monitor.launch.xml`
3. Launch diagnostics monitor and check all status are OK
   - `ros2 run rqt_runtime_monitor rqt_runtime_monitor --ros-args -r diagnostics:=diagnostics_array`

## Notes for reviewers

This PR depends on https://github.com/tier4/tier4_autoware_msgs/pull/96.

## Interface changes

None

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
